### ISSUE TITLE
Accessibility gaps found in datagrid example.

### ISSUE THREAD

**Issue by pavandixit93** [vmware/clarity#4756]
## Describe the bug

Below are the few accessibility issues found in datagrid.

1. aria-label is missing for `clr-dg-string-filter` component. `clr-dg-string-filter` can be activated in datagrid built in filters example. When user clicks on filter button `input` element is shown and aria-label is not added in `clr-dg-string-filter` component.
For you information, `clr-dg-number-field`  component has given aria-label but for `clr-dg-string-filter` aria-label is missing.
Issue can be found in https://clarity.design/documentation/datagrid/built-in-filters , Click on the filter icon button('Pokeman').

2. datagrid-action-overflow has role="menu" but when we open datagrid-action-overflow menu its children has no any role.
The childrens are mssing role="menuitem" role. this issue is for https://clarity.design/documentation/datagrid/single-action example. Click on the menu bar present before the User Id column.

## How to reproduce

_Please provide a link to a reproduction scenario using one of the [Clarity Stackblitz templates](https://stackblitz.com/@clr-team). Reports without a clear reproduction may not be prioritized until one is provided._

1.  Go to '...'
2.  Click on '....'
3.  Scroll down to '....'
4.  See error

## Expected behavior

_A clear and concise description of what you expected to happen._

## Versions

**App**

- Angular: [e.g. 6]
- Node: [e.g. 8.10.0]
- Clarity: [e.g. 0.12.5]

**Device:**

- Type: [e.g. MacBook]

## Additional notes

_Add any other notes about the problem here._

**Comment by pavandixit93** [comment]
Hi Team, Both issues are straight forward, Please let me know if any input needed from me.

**Comment by mathisscott** [comment]
@pavandixit93
Thanks for the continued interest in this issue. It is one we will be addressing but our priorities are focused on delivering 4.0 right now. If our near term objectives do not align with your product team's wants, you could submit a contribution. Updating aria attributes seems like a good candidate for helping out if you need this effort resolved quickly.

If/when you do submit a PR, though, I'd recommend pinging @chlane so that our a11y lead has a chance to vet the approach/effort. Thanks!

**Comment by chlane** [comment]
@pavandixit93 Hi pavandixit93, we've reported the first issue previously in https://github.com/vmware/clarity/issues/4664

**Comment by chlane** [comment]
@pavandixit93 Regarding the second issue, I can't find any elements with role="menu" on https://clarity.design/documentation/datagrid/single-action.  See the relevant HTML below.

```
<clr-dg-action-overflow _ngcontent-clarity-c231=""><button type="button" role="button" aria-haspopup="true" clrpopoveranchor="" clrpopoveropenclosebutton="" class="datagrid-action-toggle clr-anchor clr-smart-open-close" aria-controls="clr-action-menu7" aria-expanded="false" aria-label="Available actions"><clr-icon shape="ellipsis-vertical" title="Available actions" role="none"><svg version="1.1" class="can-badge " viewBox="0 0 36 36" preserveAspectRatio="xMidYMid meet" xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink" focusable="false" role="img" aria-labelledby="_clr_icon_16"><circle cx="18" cy="4.9" r="2.9" class="clr-i-outline clr-i-outline-path-1"></circle>
            <circle cx="18" cy="18" r="2.9" class="clr-i-outline clr-i-outline-path-2"></circle>
            <circle cx="18" cy="31.1" r="2.9" class="clr-i-outline clr-i-outline-path-3"></circle>
            <circle cx="18" cy="4.9" r="2.9" class="clr-i-outline--badged clr-i-outline-path-1--badged"></circle>
            <circle cx="18" cy="18" r="2.9" class="clr-i-outline--badged clr-i-outline-path-2--badged"></circle>
            <circle cx="18" cy="31.1" r="2.9" class="clr-i-outline--badged clr-i-outline-path-3--badged"></circle>
            <circle cx="30" cy="6" r="5" class="clr-i-outline--badged clr-i-outline-path-4--badged clr-i-badge"></circle></svg><span class="is-off-screen" id="_clr_icon_16">Available actions</span></clr-icon></button><!----></clr-dg-action-overflow>
```

**Comment by chlane** [comment]
@pavandixit93 Great catch regarding the second issue Pavan. It is valid as seen in the HTML below.

`

<div role="menu" clrfocustrap="" class="datagrid-action-overflow clr-popover-content ng-star-inserted" tabindex="0" style="top: 186.8px; left: 305px; opacity: 1;" id="clr-action-menu7" aria-hidden="false"><button _ngcontent-clarity-c231="" class="action-item">Edit</button><button _ngcontent-clarity-c231="" class="action-item">Delete</button></div>`

Fortunately, Clarity has create an accessible ARIA menu at https://clarity.design/documentation/dropdowns that can be plugged in here.

**Comment by pavandixit93** [comment]
@chlane Thank you for the update, Regarding,
<div role="menu" clrfocustrap="" class="datagrid-action-overflow clr-popover-content ng-star-inserted" tabindex="0" style="top: 186.8px; left: 305px; opacity: 1;" id="clr-action-menu7" aria-hidden="false"><button _ngcontent-clarity-c231="" class="action-item">Edit</button><button _ngcontent-clarity-c231="" class="action-item">Delete</button></div>
Actually this is part of `clr-datagrid` component so i think i can not explicitly use `clr-dropdown` there.
In above HTML `role="menu" is added but the button element are missing `role="menuitem"`
As per W3C A11Y standards, when `role="menu"` is added and if it has subsequent child elements then each child element should have child role `menuitem`.

**Comment by chlane** [comment]
@pavandixit93 understood, Clarity can use the dropdown component for that part of the Datagrid to resolve this bug.

**Comment by amellnik** [comment]
For the 2nd issue here, a simple fix would be to update all the examples to manually add `aria-role="menuitem"` to all the buttons with a note that users should do the same.  Another option would be to add something to the container that looks for all the buttons projected through `ng-content` and adds the role to each, overwriting anything that was set by the user.  Any thoughts on this?

**Comment by chlane** [comment]
aria-role="menuitem" should not be used on it's own, only a part of an ARIA menu implementation. See https://www.w3.org/TR/wai-aria-practices-1.1/#menu

**Comment by amellnik** [comment]
Hi @chlane, the wrapping element already has the `menu` role, and the linked PR adds the `menuitem` role to the child elements to fix that, so I think we should be all set on that front.

**Comment by chlane** [comment]
@amellnik Great! Please also ensure that all the expected keyboard behaviors are working as seen at  https://www.w3.org/TR/wai-aria-practices-1.1/#menu.

**Comment by pavandixit93** [comment]
Thank you team :)
